src/data/finset/basic.lean

```diff
@@ -1304,6 +1304,12 @@ theorem range_mono : monotone range := λ _ _, range_subset.2
 lemma mem_range_succ_iff {a b : ℕ} : a ∈ finset.range b.succ ↔ a ≤ b :=
 finset.mem_range.trans nat.lt_succ_iff
 
+lemma mem_range_le {n x : ℕ} (hx : x ∈ range n) : x ≤ n :=
+  (mem_range.1 hx).le
+
+lemma mem_range_sub_ne_zero {n x : ℕ} (hx : x ∈ range n) : n - x ≠ 0 :=
+by { rintros h, apply nat.le_lt_antisymm (nat.sub_eq_zero_iff_le.1 h) (mem_range.1 hx) }
```

Comment:
Also I believe that the "normal form" way to say that a natural is non-zero is `0 < n`. Maybe the conclusion should be `0 < n` in which case you could get rid of that `ne_of_gt` in my suggested proof.

Reply:
This lemma is used to dissolve a few `ite (n - x = 0) _ _` , and turns out to be quite useful in this form..

Reply:
Aah I see -- I guess if nobody else has an objection then I'm happy. There seems to be some debate in the community about what the "normal form" way of saying that a natural is non-zero is, anyway, and this is probably because \ne is sometimes useful (as you point out).